### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by janhohenheim**
- Add `--all-targets`, which is short for `--lib --bins --tests --benches --examples`
- Add `cargo doc` checks. I have found that using `cargo doc` for development can be useful to read my own documentation and understand what the heck I was thinking a month ago. As such, it would be nice to check it in the CI. I pass `--no-deps` to not compile the dependencies in that step
- Implement #122
	- Runs once for each macOS build
- Implement #118 
	- Runs only once for macOS

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -8,59 +8,60 @@ on:
 
 env:
   CARGO_TERM_COLOR: always
+  # Enable sccache to speed up builds.
+  SCCACHE_GHA_ENABLED: "true"
+  RUSTC_WRAPPER: "sccache"
```
Comment (benfrankel) [inline .github/workflows/ci.yaml:13]:
```suggestion
  SCCACHE_GHA_ENABLED: true
  RUSTC_WRAPPER: sccache
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -8,59 +8,60 @@ on:
...
 
 jobs:
-  # Run cargo test
+  # Run cargo test --all-targets
```
Comment (benfrankel) [inline .github/workflows/ci.yaml:16]:
```suggestion
  # Run unit tests.
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -8,59 +8,60 @@ on:
...
+
       - name: Install Dependencies
         run: sudo apt-get update; sudo apt-get install --no-install-recommends libasound2-dev libudev-dev
+
       - name: Run cargo test
-        run: cargo test
+        run: cargo test --all-targets
 
-  # Run cargo clippy -- -D warnings
+  # Run cargo clippy --all-targets -- -D warnings --all-targets
```
Comment (benfrankel) [inline .github/workflows/ci.yaml:40]:
```suggestion
  # Run clippy lints.
```

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -70,9 +71,41 @@ jobs:
...
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.5
+
+      - name: Cache target directory
+        uses: Leafwing-Studios/cargo-cache@v1
+
       - name: Run cargo fmt
         run: cargo fmt --all -- --check
+
+# Run cargo doc --no-deps
```
Comment (benfrankel) [inline .github/workflows/ci.yaml:89]:
```suggestion
  # Check documentation.
```

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -30,6 +30,9 @@ env:
   # Before enabling LFS, please take a look at GitHub's documentation for costs and quota limits:
   # https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-storage-and-bandwidth-usage
   USE_GIT_LFS: false
+  # Enable sccache to speed up builds.
+  SCCACHE_GHA_ENABLED: "true"
+  RUSTC_WRAPPER: "sccache"
```
Comment (benfrankel) [inline .github/workflows/release.yaml:35]:
```suggestion
  SCCACHE_GHA_ENABLED: true
  RUSTC_WRAPPER: sccache
```

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -270,6 +297,9 @@ jobs:
           toolchain: stable
           targets: x86_64-apple-darwin
```
Comment (benfrankel) [inline .github/workflows/release.yaml:299]:
```suggestion

      - name: Run sccache-cache
        uses: mozilla-actions/sccache-action@v0.0.5

```

**Review by benfrankel — CHANGES_REQUESTED** [review]
I'll trust that this works, just leaving some nits.

File: .github/workflows/release.yaml (hunk 6 above)
Comment (janhohenheim) [reply .github/workflows/release.yaml:299]: Why would you run sccache twice?

Comment (benfrankel) [reply .github/workflows/release.yaml:299]: Misread.

File: .github/workflows/ci.yaml [hunk 7]
```diff
@@ -8,59 +8,60 @@ on:
 
 env:
   CARGO_TERM_COLOR: always
+  # Enable sccache to speed up builds.
```
Comment (benfrankel) [inline .github/workflows/ci.yaml:11]:
```suggestion
  # This enables sccache to speed up builds.
```
To make it clear this isn't telling the user they need to _do something_ to enable sccache.

File: .github/workflows/release.yaml [hunk 8]
```diff
@@ -30,6 +30,9 @@ env:
   # Before enabling LFS, please take a look at GitHub's documentation for costs and quota limits:
   # https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-storage-and-bandwidth-usage
   USE_GIT_LFS: false
+  # Enable sccache to speed up builds.
```
Comment (benfrankel) [inline .github/workflows/release.yaml:33]:
```suggestion
  # This enables sccache to speed up builds.
```

**Review by benfrankel — APPROVED** [review]

**Pull request merged**